yb-voyager/src/query/sqltransformer/transformer.go

```diff
@@ -181,3 +182,40 @@ func (t *Transformer) RemoveRedundantIndexes(stmts []*pg_query.RawStmt, redundan
 
 	return sqlStmts, removedIndexToStmtMap, nil
 }
+
+func (t *Transformer) ModifySecondaryIndexesToRange(stmts []*pg_query.RawStmt) ([]*pg_query.RawStmt, []*sqlname.ObjectNameQualifiedWithTableName, error) {
+	var modifiedObjNames []*sqlname.ObjectNameQualifiedWithTableName
+	for idx, stmt := range stmts {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		if stmtType != queryparser.PG_QUERY_INDEX_STMT {
+			continue
+		}
+		indexStmt := stmt.Stmt.GetIndexStmt()
+		if indexStmt == nil {
+			continue
+		}
+		if indexStmt.AccessMethod != queryissue.BTREE_ACCESS_METHOD {
+			//In Postgres the ordered scans are only supported for btree
+			//so restricting the change to only Btree indexes
+			//refer https://www.postgresql.org/docs/current/sql-createindex.html#:~:text=For%20index%20methods%20that%20support%20ordered%20scans%20(currently%2C%20only%20B%2Dtree)%2C%20the%20optional%20clauses%20ASC
+			continue
+		}
+		if len(indexStmt.IndexParams) == 0 {
+			continue
+		}
+		//checking only the first param of the key column of the index 
+		if indexStmt.IndexParams[0].GetIndexElem() == nil {
+			continue
+		}
+		if indexStmt.IndexParams[0].GetIndexElem().Ordering != queryparser.DEFAULT_SORTING_ORDER {
```

Comment:
I don't see unit tests. I think it would be good to add a few basic unit tests. I think it should be fairly simple to do it at the transformer/sqlfiletransformer layer only)

this is a good case for a unit test (if an index already has a sorting order, we don't want to change it. )

Reply:
will take up the schema optimization ones in a separate PR. added unit tests for these transformers